source/com.microsoft.tfs.client.common.ui/src/com/microsoft/tfs/client/common/ui/protocolhandler/ProtocolHandler.java

```diff
@@ -129,55 +188,50 @@ private URI findProtocolHandlerUriArgument(final String[] applicationArgs) {
 
     /*
      * @formatter:off
-     * We're looking for a protocol handler argument among all command line
-     * arguments passed by eclipse.launcher to the Eclipse application. The
-     * protocol handler argument generated by TFS should have the following
-     * syntax:
+     * The protocol handler argument generated by TFS should have
+     * the following syntax:
      * 
-     * -clonefromtfs vsoe://<operation>/?url=<git-repository-url>&Ref=master&EncFormat=UTF8[&<other-parameters>...]
+     * -clonefromtfs vsoeclipse://checkout/?EncFormat=UTF8&tfslink=<base64 encoded parameters>
      * 
-     * At this moment we use vsoi instead of vsoe, and ignore <operation> and
-     * <other-parameters>
      * @formatter:on
      */
-    private synchronized void tryParseProtocolHandlerUri() {
+    private synchronized boolean tryParseProtocolHandlerUri() {
```

Comment:
What if you return a Boolean to indicate the operation is successful or not, and use atomicReference as a parameter to set the isAvailable flag?  Essentially an "out" parameter.